### PR TITLE
[cpptrace] Add v0.3.0

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/cpptrace/portfile.cmake [hunk 1]
```diff
@@ -2,18 +2,11 @@ vcpkg_from_github(
...
-      libdwarf_fixes.patch
-      uintptr_fix.patch
-      runtime_destination.patch
 )
 
-vcpkg_list(SET options -DCPPTRACE_USE_SYSTEM_LIBDWARF=On)
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-  vcpkg_list(APPEND options -DCPPTRACE_STATIC=On)
-endif()
+vcpkg_list(SET options -DCPPTRACE_USE_EXTERNAL_LIBDWARF=On -DCPPTRACE_VCPKG=On)
```
Comment (dg0yt) [inline ports/cpptrace/portfile.cmake:9]: Please move the options directly to `vcpkg_cmake_configure`.
And unless broken upstream, spell the boolean value as `ON`, not `On`.

Comment (jeremy-rifkin) [reply ports/cpptrace/portfile.cmake:9]: Thanks!

**Comment by nickdademo** [comment]
@jeremy-rifkin I had trouble using the new `libassert` port on Windows (VS 17.8) due to `cpptrace` compile errors - does this PR also fix that?

**Comment by jeremy-rifkin** [comment]
I'm trying to fix that right now. Cpptrace 0.3.0 has some API changes, hence the errors, but I'm not sure how to get vcpkg to use 0.2.1 for libassert.

I'm surprised you'd be seeing those errors though, since this PR hasn't been merged yet. Did you clone this branch locally or some such thing?

**Comment by nickdademo** [comment]
> I'm trying to fix that right now. Cpptrace 0.3.0 has some API changes, hence the errors, but I'm not sure how to get vcpkg to use 0.2.1 for libassert.
> 
> I'm surprised you'd be seeing those errors though, since this PR hasn't been merged yet. Did you clone this branch locally or some such thing?

Error is using current HEAD of vcpkg. So the current cpptrace port seems to be broken as well.

**Comment by jeremy-rifkin** [comment]
@nickdademo Can you open an issue on the libassert repo? I'm able to install libassert from the current head.

**Comment by jeremy-rifkin** [comment]
Regarding this PR, I could use guidance with how to handle libassert needing a specific version of cpptrace.

**Comment by dg0yt** [comment]
> Regarding this PR, I could use guidance with how to handle libassert needing a specific version of cpptrace.

You can't. The latest versions must integrate in vcpkg.

**Comment by jeremy-rifkin** [comment]
Well.... I can make a 1.2.2 release for libassert using cpptrace 0.3.0 (and delete the 1.2.1 port). I am somewhat concerned about any future API changes though.

I don't want to sound like I'm complaining but this seems like a pretty egregious shortcoming of a package manager. Out of curiosity is there a reason for not supporting this?

**Comment by dg0yt** [comment]
This is a restriction for ports. Basically it is the promise that you can install (any subset of) all ports at once. They are tested together, and they can't set conflicting version requirements.
OTOH users can ask for combining particular versions in manifest mode.
And of course packages can check the versions of dependencies during the build,
Don't discuss with me, I'm just explaining,,,

> and delete the 1.2.1 port

... update the port to 1.2.2. Or add a patch.

**Comment by jeremy-rifkin** [comment]
Thanks for the explanation

**Comment by jeremy-rifkin** [comment]
A couple checks are still queued but I think I’ve sorted everything out. Thanks again for the help :)

**Review by LilyWangLL — APPROVED** [review]

**Comment by ndk** [comment]
I suggest adopting a more consistent naming convention for files. Specifically, the cmake configs are installed in a folder named `libassert,` yet the config names lack the `lib` prefix, which is also absent from the library name. This inconsistency makes it challenging to use the library directly by adding `find_package(libassert REQUIRED)` in `CMakeLists.txt`. At least, it happens with VC++2022.

Additionally, it might be beneficial to add a usage hint. For an example, please refer to this: https://github.com/microsoft/vcpkg/blob/master/ports/openal-soft/usage

**Comment by jeremy-rifkin** [comment]
@ndk Thank you and apologies for the inconvenience! This pr is for cpptrace but that stuff for libassert I think is already taken care of on the v2 dev branch. Another contributor helped me overhaul the cmake there and I'm hoping to release stuff in a couple weeks. A `usage` file is already specified for this port. If anything else comes up, please let me know and thank you for your interest in the library :) (It's probably best to open issues on the library repositories themselves)

**Pull request merged**